juniper_official/Protocols/Fib/fib-summary.rule

```diff
@@ -0,0 +1,162 @@
+/*
+ * Detects forwarding-table of each route table and its protocol's route count
+ * threshold breaches and notify anomaly when route count is aberrant.
+ */
+ iceberg {
+    topic protocol.routesummary {
+        description "Monitors each forwarding table and each protocol route count and sets dynamic thresholds and notify anomaly when route count is abnormal";
+        synopsis "FIB route tables and protocols statistics analyzer";    
+        rule check-fib-summary {
+            synopsis "Forwarding-table protocols routes statistics analyzer";
+            description "Collects forwarding-table's total-route-count of each protocol and sets dynamic thresholds and notify anomaly when route count is abnormal";
+            /*
+             * Monitors forwarding-table's each protocol's route. Sets dynamic
+             * threshold using k-means machine learning alogorithm based on
+             * learned data. Notifies via dashboard colors when the route
+             * count of each protocol exceed static threshold for every
+             * collected metric during a 5 minute time range (red). Stes coloe
+             * to yellow when route-count exceed dynamic threshold. Otherwise
+             * the color is set to green.
+             * 
+             * Use address-family enabled-protocols logical-system-name route-table-type and table-name name as keys for rule.
+             */
+             keys [ address-family enabled-protocols logical-system-name route-table-type table-name ];
+            /*
+             * Sensor configuration to get data from network devices
+             */            
+             sensor fib-sensor {
+                synopsis "FIB iAgent sensor definition";
+                description "iAgent sensor collect forwarding-table route-count stats from network device";
+                 iAgent {
+                     file fib.yml;
+                     table FibSummaryTable;
+                     frequency 15m;
+                 }
+             }
+            /*
+             * Fields defined with dynamic threshold ML algorithms
+             */
+             field dt-route-count {
+                 formula {
+                     dynamic-threshold {
+                         algorithm k-means;
+                         learning-period 1m;
```

Comment:
Given that our default retention policy is 7 days, this combination of pattern-periodicity and learning-period won't work. Can we bump down the pattern-periodicity to 1d or 1d1h and the learning period to 7 days?

Also, can you please update to the new syntax and not use the deprecated syntax. Though, I am not sure whether the code is in yet. :(
But, can you check with Mohit on what he plans to introduce?

Reply:
Modified and committed code

Reply:
Modified and committed code